tests/Avalonia.Controls.UnitTests/ListBoxTests_Single.cs

```diff
@@ -177,6 +180,71 @@ public void Setting_Item_IsSelected_Sets_ListBox_Selection()
             Assert.Equal(1, target.SelectedIndex);
         }
 
+        private class TestDataContext : ReactiveObject
+        {
+            private int _counter;
+            private string _selectedItem;
+
+            public TestDataContext()
+            {
+                Items1 = new ObservableCollection<string>(Enumerable.Range(1, 5).Select(i => GenerateItem()));
```

Comment:
It's a small thing, but this took me a while to grok: rather than using a `_counter` field and a separate method feels like it would be clearer to go:

```
Items1 = new ObservableCollection<string>(Enumerable.Range(0, 5).Select(i => $"Item {i}"));
Items2 = new ObservableCollection<string>(Enumerable.Range(5, 5).Select(i => $"Item {i}"));
```

Or maybe it's just me?

Also another nit: rather than breaking up the flow of the unit tests, related nested test classes are put at the bottom in other unit test classes.